Src/AmrCore/AMReX_FillPatchUtil_I.H

```diff
@@ -573,15 +573,25 @@ namespace {
 
                 for (int d=0; d<AMREX_SPACEDIM; ++d)
                 {
-                    mf[d]->ParallelCopy(mf_refined_patch[d], 0, dcomp, ncomp, IntVect{0}, nghost);
+                    bool aliasing = false;
+                    for (auto const& fmf_a : fmf) {
+                        aliasing = aliasing || (mf[d] == fmf_a[d]);
+                    }
```

Comment:
This is directly comparing the pointers, correct? Is that sufficient? That this will likely only trigger when the function is called with explicitly identical pointers? Any other cases we want to catch?

Also, do we expect cases where `mf[0] != fmf[0] && mf[0] == fmf[1]`? The nested AMREX_SPACEDIM loops feel off and like they might confuse things if we take a look later. (This beckons back to our discussion about what `ASSERT`s are appropriate for this function.)

Reply:
This is sufficient unless people use alias MultiFabs in `mf` or `fmf`.  In that case, we will have to compare the data pointers to the data.  We could do that, but we could argue that's very unlikely to happen.

I think it's possible `mf[0] != fmf[0] && mf[0] == fmf[1]`.

Reply:
I think this is ready to go if we've decided that we aren't concerned with aliases at this point. I'll approve and you can decide. 😃